sdk/assemblyscript/src/assembly/dynamicmap.ts

```diff
@@ -63,303 +50,17 @@ export class DynamicMap {
     return this.data.values();
   }
 
-  __INITIALIZE(): this {
-    return this;
-  }
-
-  __SERIALIZE(): string {
-    // This would be ideal, but doesn't work:
-    //   return JSON.stringify(this.data);
-    // So instead, we have to do construct the JSON manually.
-    //
-    // TODO: Update this to use JSON.stringify once it's supported in json-as.
-    // https://github.com/JairusSW/as-json/issues/98
-
-    const segments: string[] = [];
-    const keys = this.data.keys();
-    const values = this.data.values();
-
-    for (let i = 0; i < this.data.size; i++) {
-      const key = JSON.stringify(keys[i]);
-      const value = values[i]; // already in JSON
-      segments.push(`${key}:${value}`);
-    }
 
-    return `{${segments.join(",")}}`;
+  @serializer
+  serialize(self: DynamicMap): string {
```

Comment:
I think `serialize` and `deserialize` custom methods can be marked `private` and still work correctly.  That would keep them out of autocomplete for the user.

(applies throughout)